spec/statement.md

```diff
@@ -0,0 +1,131 @@
+consumed line.
+
+A document parsed in the indented syntax may not begin with whitespace, and the
+initial `current indentation level` for any document is `0`.
+
+Changes in the indentation level are used by the indented syntax to start and
+end blocks of statements.
+
+[`documentation indentation character`]: #document-indentation-character
+
+### Document indentation character
+
+The `document indentation character` is the character used for calculating the
+[`current indentation level`], and is either a space or tab character.
+
+[`current indentation level`]: #current-indentation-level
+
+It is defined as the first tab or space character used as an [IndentMore]
+production in a document.
+
+In the indented syntax, no character other than the `document indentation
+character` may be used for indentation.
+
+## Syntax
+
+### ScssStatements
+
+<x><pre>
+**ScssStatements**      ::= (Statement ';'?¹)*Statement?
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment` or `SilentComment`, or after the final production in a
+`ScssStatements` production, which is either at the end of a `Block` or the end
+of the document.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement IndentedSame?¹)* Statement?
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment` or `SilentComment`, or after the final production in a
+`IndentedStatements` production, which is either at the end of a `Block` or the
+end of the document.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`IndentedStatements` rule, parse it preferentially as a `Statement`.
+
+### Statements
+
+<x><pre>
+**Statements**          ::= [ScssStatements] | [IndentedStatements]¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements] [IndentLess]
+**Block**          ::= ScssBlock | IndentedBlock¹
+</pre></x>
+
+[IndentMore]: #indentation
+[IndentLess]: #indentation
+
+1: Only the production for the current syntax is valid.
+
+### Whitespace
+
+> Whitespace separates productions inside a statement when disambiguation is
+> necessary. The Whitespace productions do not separate Statements.
+
+<x><pre>
+**LineBreak**               ::= CarriageReturn | LineFeed | FormFeed
+**ScssWhitespace**          ::= LineBreak | Space | Tab
+**IndentedWhitespace**      ::= Space | Tab
+</pre></x>
+
+### Indentation
+
+<x><pre>
+**IndentCharacter**         ::= Space | Tab
+**IndentSame**              ::= [LineBreak] [IndentCharacter]{ Current }
+**IndentLess**              ::= [LineBreak] [IndentCharacter]{ 0, Current - 1 }
```

Comment:
If this consumes the indentation, then there won't be anything for the next statement to consume as `IndentSame`. The end of the document should also be considered an `IndentLess`.